apps/site/assets/ts/stop/components/DepartureTimes.tsx

```diff
@@ -3,33 +3,329 @@ import usePredictionsChannel from "../../hooks/usePredictionsChannel";
 import { isACommuterRailRoute } from "../../models/route";
 import { DirectionId, Route, Stop } from "../../__v3api";
 import renderFa from "../../helpers/render-fa";
-import { formatDepartureTime } from "../../helpers/date";
+import { formatToBostonTime } from "../../helpers/date";
+import { filter, find, groupBy, slice } from "lodash";
+import {
+  differenceInSeconds,
+  isSameDay,
+  secondsInDay,
+  secondsInHour,
+  secondsInMinute
+} from "date-fns";
+import realtimeIcon from "../../../static/images/icon-realtime-tracking.svg";
+import SVGIcon from "../../helpers/render-svg";
+import { ScheduleWithTimestamp } from "../../models/schedules";
+import { PredictionWithTimestamp } from "../../models/perdictions";
 
 interface DepartureTimesProps {
   route: Route;
   stop: Stop;
   directionId: DirectionId;
+  schedulesForDirection: ScheduleWithTimestamp[];
 }
 
+// This interface is used to tell the front end
+// how to display the ScheduleInfoModel data
+interface DisplayTimeConfig {
+  isPrediction?: boolean;
+  displayString: string;
+  trackName?: string | null;
+  isTomorrow?: boolean;
+  isBolded?: boolean;
+  isStikethrough?: boolean;
```

Comment:
I was trying to do what you suggesting which is have one function determine what times to display, and how to display them (`infoToDisplayTime`), and a second function to actually build the display component (`departureTimeRow`).  They were named this way because those are the css options those variables control, and it gave granular control over all the states.

The times should be bolded if it is the first one in the display.  The times should be striked through if it is the second time in a list and the time is either delayed or cancelled.  The reason that bolded isn't based off position in list and instead uses a variable is that `infoToDisplayTime` can return the strings `Updates unavailable` or `No times in 24 hours` which should not be bolded.

I can change these names to something else, but I felt like they fit because they describe exactly how the display string will be modified.